### ISSUE TITLE
Some reflection/construction code needs tests

### ISSUE THREAD

**Issue by tedinski** [melt-umn/silver#318]
I just discovered that the `silver:extensions:silverconstruction` code uses full names in strings like `"silver:extension:silverconstruction:makeQName"`, but this function was actually recently moved to `silver:hostEmbedding`. So this is currently broken, I assume.

But this wasn't caught, which means nothing is testing these. These should have tests somewhere, somehow.

I'm going to change these strings now, but I'm not going to test if my fix worked. Someone write some tests for this eventually, plz, thx.

**Comment by tedinski** [comment]
You know what, I think this code was also supplying arguments in the wrong order, so I think it was never run at all. :)

**Comment by krame505** [comment]
Probably.  I think this extension is currently only used in a couple of places at the moment.  

Not sure about the best option for testing.  We should somehow be testing the `silver:hostEmbedding` library, but I suppose direct tests for the `silverconstruction` extension could be sufficient.  I guess we would just need to explicitly construct some Silver `Exprs` using this extension, `hackUnparse` them, and compare the result to what we would expect.

**Comment by tedinski** [comment]
The thing we need to ensure is that we're actually testing it works. If that makes sense.

That is, we don't want to end up comparing trees, because that would just be a brittle test, and it wouldn't have spotted this bug: we'd still be generating the old (now wrong) full name, and asserting against the (now wrong) "correct value," which would continue to pass.

We want to take that generated code and have it get compiled, so name binding would go "hey this name here doesn't exist."

**Comment by krame505** [comment]
Makes sense.  This sort of testing really requires us to use this extension in implementing other Silver extensions, and testing that those extensions forward to something that compiles.  
We could go and add a dummy "test" extension to Silver that does this, but perhaps a better option is to just try and use all the features of the construction extension by replacing annoying pieces of AST construction code in existing Silver extensions.
